addons/l10n_in/models/account.py

```diff
@@ -10,6 +10,16 @@ class AccountMoveLine(models.Model):
     _inherit = "account.move.line"
 
     l10n_in_hsn_code = fields.Char(string="HSN/SAC Code", compute="_compute_l10n_in_hsn_code", store=True, readonly=False, copy=False)
+    is_valid_l10n_in_hsn_code = fields.Boolean(compute="_compute_is_valid_l10n_in_hsn_code")
```

Comment:
Seems overkill to have the lines turn red.  We could also use the new actionable_errors widget for the banner and go to a specific lists of products where the hsn fields are visible (not sure if the autocomplete would work in a tree view)